CMakeLists.txt

```diff
@@ -266,6 +266,7 @@ find_package(FLEX REQUIRED)
 find_package(Readline REQUIRED)
 find_package(NCURSES REQUIRED)
 find_package(LibLZMA MODULE)
+find_package(CURL REQUIRED)
```

Comment:
Thanks for your contribution. 
If you want to introduce new third party libs, we need to repackage our third-party.rpm.  

Of course, i like the way using the libs directly.

Reply:
> Of course, i like the way using the libs directly.
+1

Reply:
CURL is one of common libraries provided by many Linux distributions. Why don't we make full use of these system-provided libraries?

Reply:
Indeed, that is.  
But nebula's all libraries are recompiled by higher gcc version (gcc-7.x above). I am not sure the system-provided  one works well,  have you tested it in your env?

Reply:
In my env(CentOS 6), system-provided CURL shows good compatibility and stability. Nebula works well with it.

Reply:
CURL and some other system-provided libs(zlib, bzip2, libevent, etc) are written in C. Whether compiled by higher or lower gcc, they will finally be linked with system-provided glibc(2.12 on CentOS6). So i make a suggestion that nebula can use some of system-provided libs if they meet the requirements of nebula, fbthrift, folly, etc.

Reply:
Currently,  almost all libs depended by nebula are recompiled includes "zlib, bzip2, zstd, libevent, openssl" etc. Even the "ld" are replaced by ourselves in some Env(For example, centos 6.x).    

I notice libcurl has some dependencies such as openssl.  
To be safe, i suggest we implement the build script under third-party/externals likes other libs. 

wdyt?  @GestureWei

Reply:
I think we can use this way to use libcurl

Reply:
That's ok.
Don't block the improvement by it.  Let's go on.